classes/Commands/UpdateCommand.php

```diff
@@ -176,6 +179,46 @@ private function processConsoleInputConfiguration(InputInterface $input): void
         }
     }
 
+    /**
+     * @throws DistributionApiException
+     * @throws UpgradeException
+     * @throws Exception
+     */
+    private function calculateUpdateTypeAfterConfigLoad(): void
+    {
+        $updateConfiguration = $this->upgradeContainer->getUpdateConfiguration();
+        $channel = $this->consoleInputConfiguration[UpgradeConfiguration::CHANNEL] ?? $updateConfiguration->getChannel();
+        $currentVersion = $this->upgradeContainer->getProperty(UpgradeContainer::PS_VERSION);
+        $destinationVersion = null;
+
+        switch ($channel) {
+            case UpgradeConfiguration::CHANNEL_LOCAL:
+                $zipFile = $this->consoleInputConfiguration[UpgradeConfiguration::ARCHIVE_ZIP] ?? $updateConfiguration->getLocalChannelZip();
+                if ($zipFile) {
+                    $fullFilePath = $this->upgradeContainer->getProperty(UpgradeContainer::DOWNLOAD_PATH) . DIRECTORY_SEPARATOR . $zipFile;
+                    try {
+                        $destinationVersion = $this->upgradeContainer->getPrestashopVersionService()->extractPrestashopVersionFromZip($fullFilePath);
+                        $updateConfiguration->set(UpgradeConfiguration::ARCHIVE_VERSION_NUM, $destinationVersion);
+                    } catch (Exception $e) {
+                        $this->logger->error('Unable to extract PrestaShop version from ZIP file: ' . $e->getMessage());
```

Comment:
Does it block anything? Otherwise:

```suggestion
                        $this->logger->warning('Unable to extract PrestaShop version from ZIP file, cannot compute update type: ' . $e->getMessage());
```